sdk/privatedns/Azure.ResourceManager.PrivateDns/tests/Scenario/PrivateDnsTests.cs

```diff
@@ -174,6 +189,47 @@ public async Task AddRemoveTag(bool? useTagResource)
             Assert.AreEqual(0, privateZone.Data.Tags.Count);
         }
 
+        [RecordedTest]
+        public async Task AddVnetLinkWithResolutionPolicy()
+        {
+            string privateZoneName = "privatelink.sql.azuresynapse.net";
+            string vnetName = "samplevnet";
+            string vnetLinkName = $"{Recording.GenerateAssetName("samplelink")}";
+            var privateZone = await CreatePrivateZone(_resourceGroup, privateZoneName);
+            var vnet = await CreateVirtualNetwork(_resourceGroup, vnetName);
+
+            // Add NxDomainRedirect resolutionPolicy
+            var virtualNetworkLinkData = new VirtualNetworkLinkData("global")
+            {
+                VirtualNetwork = new WritableSubResource { Id = vnet.Id },
+                RegistrationEnabled = false,
+                ResolutionPolicy = ResolutionPolicy.NxDomainRedirect,
+            };
+
+            await CreateOrUpdateVirtualNetworkLink(privateZone, vnetLinkName, virtualNetworkLinkData);
+
+            if (TestEnvironment.Mode == RecordedTestMode.Record)
+            {
+                Thread.Sleep(30000);
+            }
+
+            var virtualNetworkLinkResource = await privateZone.GetVirtualNetworkLinks().GetAsync(vnetLinkName);
+            var vnetLinkData = virtualNetworkLinkResource.Value.Data;
+            Assert.AreEqual(vnetLinkData.ResolutionPolicy, virtualNetworkLinkData.ResolutionPolicy);
+
+            // Update resolutionPolicy to default
+            virtualNetworkLinkData.ResolutionPolicy = ResolutionPolicy.Default;
+            await CreateOrUpdateVirtualNetworkLink(privateZone, vnetLinkName, virtualNetworkLinkData);
+            if (TestEnvironment.Mode == RecordedTestMode.Record)
```

Comment:
Just curious what is the behaviour for "record" here. I think the other enum values are "live" and "playback".

Reply:
Record is similar to Live but it records the requests/responses in a json file (which need to be uploaded to a different repo), which is then used for Playback mode. Playback mode does not make actual network calls, and uses the json to simulate request/response.